sky/provision/azure/instance.py

```diff
@@ -23,7 +21,9 @@
 
 if typing.TYPE_CHECKING:
     from azure.mgmt import compute as azure_compute
-    from azure.mgmt import resource as azure_resource
+    from azure.mgmt.compute import models as azure_compute_objs
+    from azure.mgmt.network import models as azure_network_objs
```

Comment:
minor nits: following the naming convention will make it a bit easier to read. Also, `objs` is more about an instance of a class in python, while here it is more a model, see pydantic.BaseModel.
```suggestion
    from azure.mgmt.compute import models as azure_compute_models
    from azure.mgmt.network import models as azure_network_models
```